src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -571,7 +571,12 @@ private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (Nullabili
             var compilation = binder.Compilation;
             if (compilation.LanguageVersion < MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion() || !compilation.ShouldRunNullableWalker)
             {
+#if DEBUG
+                // Run analysis even though it's not needed
```

Comment:
I'd add a bit more context here about "why" this is being done. Perhaps even link to issues that would've been caught in our own testing had this been enabled from the start.

Reply:
This is only going to affect compile-time analysis of attributes, which is a fairly narrow change. In debug scenarios, we're already running the nullable walker unconditional for every other scenario, including in calls to the `SemanticModel`.

---
In reply to: [355806363](https://github.com/dotnet/roslyn/pull/40261#discussion_r355806363) [](ancestors = 355806363)